### DISCUSSION TITLE
How to know how many retained messages are in a broker.

### DISCUSSION THREAD

**Discussion by pdufrene** [dotnet/MQTTnet#1575] [Q&A]
Hi,
I am using MqttClient class to connect to external Mqtt Broker.
On this Broker, are published several messages with retained flag inside root/topic1, root/topic2, root/topic3, ... 
By default I don't know how many topics are under root.

I would like to retrieve all topics payload, so I subscribe to root/#, that's fine.
Is there a way to know/detect when I have received all payloads ?
Said differently, is there a way to know how many topics are below the topic root ?

**Comment by gfoidl** [comment]
In general a broker / server should publish [SYS Topics](https://github.com/mqtt/mqtt.org/wiki/SYS-Topics) -- see that list for the information you need.

> external Mqtt Broker

It may depend on the concrete product you are connection too if the desired info as available or published.
